unstructured/ingest/v2/processes/connectors/salesforce.py

```
@@ -0,0 +1,293 @@
+            to_email=email_json.get("ToAddress"),
+            textbody=email_json.get("TextBody"),
+            htmlbody=email_json.get("HtmlBody"),
+        )
+        return dedent(eml)
+
+    @SourceConnectionNetworkError.wrap
+    def _get_response(self, file_data: FileData):
+        client = self.connection_config.get_client()
+        return client.query(
+            f"select FIELDS(STANDARD) from {file_data.additional_metadata['record_type']} where Id='{file_data.identifier}'",  # noqa: E501
+        )
+
+    def get_record(self, file_data: FileData) -> OrderedDict:
+        # Get record from Salesforce based on id
+        response = self._get_response(file_data)
+        logger.debug(f"response was returned for salesforce record id: {file_data.identifier}")
+        records = response["records"]
+        if not records:
+            raise ValueError(
+                f"No record found with record id {file_data.identifier}: {json.dumps(response)}"
+            )
+        record_json = records[0]
+        return record_json
+
+    def run(self, file_data: FileData, **kwargs: Any) -> DownloadResponse:
+        record = self.get_record(file_data)
+
+        try:
+            if file_data.additional_metadata["record_type"] == "EmailMessage":
+                document = self._eml_for_record(record)
+            else:
+                document = self._xml_for_record(record)
+            download_path = self.get_download_path(file_data=file_data)
+            download_path.parent.mkdir(parents=True, exist_ok=True)
+
+            with open(download_path, "w") as page_file:
+                page_file.write(document)
+
+            return DownloadResponse(
+                file_data=file_data,
+                path=Path(download_path),
+            )
+
+        except Exception as e:
+            logger.error(
+                f"Error while downloading and saving file: {file_data.identifier}.",
+            )
+            logger.error(e)
```

We should reraise something here so that the error gets caught intentionally rather than simply logging it and letting the process swallow it up. 

```suggestion
            logger.error(
                f"Error while downloading and saving file {file_data.identifier}: {e}",
            )
            raise e
```